quickstarts/microsoft.network/vnet-create-with-diagnostic-logs/main.bicep

```diff
@@ -0,0 +1,129 @@
+@description('Virtual network name')
+param vnetName string = uniqueString(resourceGroup().id)
+
+@description('Virtual network location')
+param location string = resourceGroup().location
+
+@description('Array containing virtual network address space(s)')
+param vnetAddressSpace array = [
+  '10.0.0.0/16'
+]
+
+@description('Array containing DNS Servers')
+param dnsServers array = []
+
+@description('Array containing subnets to create within the Virtual Network')
+@metadata({
+  name: 'Subnet name'
+  addressPrefix: 'Subnet address prefix'
+  delegation: 'The name of the service to whom the subnet should be delegated (e.g. Microsoft.Web/serverFarms)'
+  natGatewayId: 'Resource id of the NAT gateway to associate to subnet'
+  nsgId: 'Resource id of the Network Security Group to associate to subnet'
+  udrId: 'Resource id of the Route Table to associate to subnet'
+  privateEndpointNetworkPolicies: 'Enable or disable Private Endpoint network policies on subnet'
+  privateLinkServiceNetworkPolicies: 'Enable or disable PrivateLink service network policies on subnet'
+  serviceEndpoints: [
+    {
+      service: 'The type of the endpoint service'
+    }
+  ]
+})
+param subnets array = [
+  {
+    name: 'subnet1'
+    addressPrefix: '10.0.1.0/24'
+    privateEndpointNetworkPolicies: 'disabled'
+    privateLinkServiceNetworkPolicies: 'disabled'
+    delegation: null
```

Comment:
let's remove the null values (or populate them) - will allow for a better test

Reply:
cant remove the null values as it will result in a property doesn't exist error. Populated the null values i could. 
The properties requiring resource ids remain as null.

Reply:
you have to wrap them in a contains check, e.g.

```
contains(subnet, 'nsgId') ? empty(nsgId) ? null : ...

Reply:
added contains to subnet array so all properties no longer need to be specified.